spec/calendar.html

```diff
@@ -5,36 +5,83 @@
   <h1>Temporal.Calendar Objects</h1>
   <p>A Temporal.Calendar object is an Object representing a calendar.</p>
 
+  <emu-clause id="sec-calendar-types">
+    <h1>Calendar Types</h1>
+
+    <emu-note type="editor">
+      <p>
+        This section shares some text with the <a href="https://tc39.es/proposal-intl-enumeration/">Intl.Enumeration proposal</a>.
+      </p>
+    </emu-note>
+
+    <p>
+      At a minimum, ECMAScript implementations must support a built-in calendar named *"iso8601"*, representing the ISO 8601 calendar.
+      In addition, implementations may support any number of other built-in calendars.
+    </p>
+    <p>
+      ECMAScript implementations identify built-in calendars using a <dfn variants="calendar types">calendar type</dfn> as defined by <a href="https://unicode.org/reports/tr35/tr35-dates.html#Calendar_Elements">Unicode Technical Standard #35, Part 4, Section 2</a>.
+      Their canonical form is a string containing all lower case letters with zero or more hyphens.
+    </p>
+    <p>
+      <emu-xref href="#sec-temporal-calendar-constructor">The `Temporal.Calendar` constructor</emu-xref>, when called with the name of a built-in calendar as the argument, will return a valid `Temporal.Calendar` object.
+      When called with any other string, it will throw a *RangeError* exception.
+    </p>
+
+    <emu-clause id="sec-temporal-isbuiltincalendar" type="abstract operation">
+      <h1>
+        IsBuiltinCalendar (
+          _id_: a String,
+        ): a Boolean
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>The returned value is *true* if _id_ is a calendar type denoting a built-in calendar, and *false* otherwise.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _calendars_ be AvailableCalendars().
+        1. If _calendars_ contains _id_, return *true*.
+        1. Return *false*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-availablecalendars" type="abstract operation">
+      <h1>
+        AvailableCalendars (
+        ): a List of Strings
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>The returned List contains a single calendar type, *"iso8601"*.</dd>
+      </dl>
+
+      <p>
+        An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the AvailableCalendars abstract operation as specified in the ECMA-402 specification.
+        If an ECMAScript implementation does not include the ECMA-402 API the following specification of the AvailableCalendars abstract operation is used.
+      </p>
+      <emu-alg>
+        1. Return « *"iso8601"* ».
+      </emu-alg>
```

Comment:
This seems to require that an implementation not including ECMA-402 must support _only_ "iso8601", which is a constraint not shared by any other Intl extension point. Is it intentional?
```suggestion
      <dl class="header">
      </dl>

      <p>
        An ECMAScript implementation that includes the ECMA-402 Internationalization API must implement the AvailableCalendars abstract operation as specified in the ECMA-402 specification.
        If an ECMAScript implementation does not include the ECMA-402 API the following specification of the AvailableCalendars abstract operation is used.
      </p>
      <p>
        This function returns a List of String values representing calendar types supported by the implementation. The contents of the List are implementation-defined, but must contain *"iso8601"* and must not contain any String value that does not identify a calendar type in the <a href="https://cldr.unicode.org/">Unicode Common Locale Data Repository (CLDR)</a>.
      </p>
```

Reply:
That was intentional. I know we have a different situation for time zones but I felt it would be best to avoid the problem that made us decide on #1996, if possible.

This refactor should make it easier to change that if need be; we'd just use language similar to that of time zones. However, I think that might require a normative change.

I'm not sure what extension points you're referring to though — I think Temporal's time zones and calendars are the only places where we have these otherwise Intl-only concepts entering into ECMA-262?

Reply:
> That was intentional. I know we have a different situation for time zones but I felt it would be best to avoid the problem that made us decide on #1996, if possible.
> 
> This refactor should make it easier to change that if need be; we'd just use language similar to that of time zones. However, I think that might require a normative change.

Good point; opened #2160.

> I'm not sure what extension points you're referring to though — I think Temporal's time zones and calendars are the only places where we have these otherwise Intl-only concepts entering into ECMA-262?

`toLocaleString` and other "locale" methods (on e.g. [Number](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-number.prototype.tolocalestring), [BigInt](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-bigint.prototype.tolocalestring), [Date](https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-date.prototype.tolocalestring), [String](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.localecompare), and [Array](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-array.prototype.tolocalestring)) all allow implementation-defined behavior in the absence of ECMA-402.

Reply:
This definitely doesn't seem right to me - hosts are empowered to do anything Intl can do, and this limits that power.

Reply:
I'll comment over at #2160.